vsix/CppWinRTPackageTemplateWizard.cs

```diff
@@ -0,0 +1,122 @@
+        {
+            string text = $"{ex.GetType().FullName}: {ex.Message}\r\n\r\n{ex.StackTrace}";
+            MessageBox.Show(text, "Caught Exception", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            throw new WizardBackoutException();
+        }
+
+        private void InitializeWizard(DTEProject automationProject)
+        {
+            try
+            {
+                VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
+                Project project = new Project(automationProject.FullName);
+                if (project.GetPropertyValue("CppWinRTDisableAutoNuGetReference").Equals("true", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Do nothing, the NuGet integration has been explicitly disabled.
+                }
+                else
+                {
+                    Assembly asm = Assembly.Load("NuGet.VisualStudio.Interop, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
+                    Type wizardType = asm.GetType("NuGet.VisualStudio.TemplateWizard");
+
+                    wizardImpl = (IWizard)Activator.CreateInstance(wizardType);
+                }
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void BeforeOpeningFile(DTEProjectItem projectItem)
+        {
+            try
+            {
+                wizardImpl?.BeforeOpeningFile(projectItem);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void ProjectFinishedGenerating(DTEProject project)
+        {
+            try
+            {
+                // This is duplicated here to silence a warning.
+                VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
+                InitializeWizard(project);
+                wizardImpl?.ProjectFinishedGenerating(project);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void ProjectItemFinishedGenerating(DTEProjectItem projectItem)
+        {
+            try
+            {
+                wizardImpl?.ProjectItemFinishedGenerating(projectItem);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void RunFinished()
+        {
+            try
+            {
+                wizardImpl?.RunFinished();
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void RunStarted(object automationObject, Dictionary<string, string> replacementsDictionary, WizardRunKind runKind, object[] customParams)
+        {
+            // Unfortunately, we cannot forward this call to the NuGet template wizard because we can't get the EnvDTE.Project right now.
```

Comment:
I assume that limitation is benign?

Reply:
The NuGet wizard implementation [does run some code in RunStarted()](https://github.com/NuGet/NuGet.Client/blob/f67a6083f8a49502f52f0c50492db03452785359/src/NuGet.Clients/NuGet.VisualStudio.Implementation/VsTemplateWizard.cs#L317-L339). However, the packages don't actually get installed [until ProjectFinishedGenerating() is called](https://github.com/NuGet/NuGet.Client/blob/f67a6083f8a49502f52f0c50492db03452785359/src/NuGet.Clients/NuGet.VisualStudio.Implementation/VsTemplateWizard.cs#L260-L290). I tested this PR locally, and it does work. That being said, I can certainly refactor this code to call RunStarted() unconditionally; I don't think that would harm anything.

Reply:
This was done.